src/stats/cgroup_v1/mod.rs

```diff
@@ -0,0 +1,203 @@
+use crate::stats::{CpuUsageValue, SystemStatsSource};
+mod cpu_usage;
+mod memory_current;
+mod memory_max;
+mod num_cpus;
+use std::fs::File;
+use std::io::{self, BufRead, BufReader};
+use std::path::PathBuf;
+
+#[cfg(test)]
+use mockall::automock;
+
+#[derive(Default, Clone)]
+pub struct CgroupV1MountPoints {
+    pub cpu: Option<PathBuf>,
+    pub cpuacct: Option<PathBuf>,
+    pub memory: Option<PathBuf>,
+}
+
+pub struct CgroupV1Source<P: CgroupV1Provider> {
+    provider: P,
+}
+
+impl<P: CgroupV1Provider> CgroupV1Source<P> {
+    fn new(provider: P) -> Self {
+        Self { provider }
+    }
+}
+
+impl CgroupV1Source<CgroupV1FilesystemReader> {
+    pub fn with_filesystem_reader_at(mount_points: CgroupV1MountPoints) -> Self {
+        Self::new(CgroupV1FilesystemReader::new(mount_points))
+    }
+}
+
+impl<P: CgroupV1Provider> SystemStatsSource for CgroupV1Source<P> {
+    fn get_num_cpus(&self) -> io::Result<f64> {
+        num_cpus::get_num_cpus(&self.provider)
+    }
+
+    fn get_cpu_usage(&self) -> io::Result<CpuUsageValue> {
+        cpu_usage::get_cpu_usage(&self.provider)
+    }
+
+    fn get_memory_usage_kb(&self) -> io::Result<u64> {
+        memory_current::get_memory_usage_kb(&self.provider)
+    }
+
+    fn get_memory_total_kb(&self) -> io::Result<u64> {
+        memory_max::get_memory_max_kb(&self.provider)
+    }
+}
+
+pub struct CgroupV1FilesystemReader {
+    mount_points: CgroupV1MountPoints,
+}
+
+impl CgroupV1FilesystemReader {
+    fn new(mount_points: CgroupV1MountPoints) -> Self {
+        Self { mount_points }
+    }
+
+    fn cpu_quota_path(&self) -> Option<PathBuf> {
+        self.mount_points
+            .cpu
+            .as_ref()
+            .map(|pb| pb.join("cpu.cfs_quota_us"))
+    }
+
+    fn cpu_period_path(&self) -> Option<PathBuf> {
+        self.mount_points
+            .cpu
+            .as_ref()
+            .map(|pb| pb.join("cpu.cfs_period_us"))
+    }
+
+    fn cpu_usage(&self) -> Option<PathBuf> {
+        self.mount_points
+            .cpuacct
+            .as_ref()
+            .map(|pb| pb.join("cpuacct.usage"))
+    }
+
+    fn memory_usage_path(&self) -> Option<PathBuf> {
+        self.mount_points
+            .memory
+            .as_ref()
+            .map(|pb| pb.join("memory.usage_in_bytes"))
+    }
+
+    fn memory_limit_path(&self) -> Option<PathBuf> {
+        self.mount_points
+            .memory
+            .as_ref()
+            .map(|pb| pb.join("memory.limit_in_bytes"))
+    }
+
+    fn memory_stat_path(&self) -> Option<PathBuf> {
+        self.mount_points
+            .memory
+            .as_ref()
+            .map(|pb| pb.join("memory.stat"))
+    }
```

Comment:
Feels like these should be precomputed based on `mount_points`?

Reply:
yup! they could be